datastore/mongo/api_key.go

```diff
@@ -65,8 +65,8 @@ func (db *apiKeyRepo) FindAPIKeyByMaskID(ctx context.Context, maskID string) (*d
 	apiKey := new(datastore.APIKey)
 
 	filter := bson.M{
-		"mask_id":         maskID,
-		"document_status": datastore.ActiveDocumentStatus,
+		"mask_id":    maskID,
+		"deleted_at": nil,
```

Comment:
The store already handles this, do we still need to pass it here?

Reply:
I think this affects almost all repository methods.